### ISSUE TITLE
PVS-Studio: misprint in dh_cmp_parameters()

### ISSUE THREAD

**Issue by AndreyKarpov** [openssl/openssl#16562]
```
static int dh_cmp_parameters(const EVP_PKEY *a, const EVP_PKEY *b)
{
    return ossl_ffc_params_cmp(&a->pkey.dh->params, &a->pkey.dh->params,
                               a->ameth != &ossl_dhx_asn1_meth);
}
```

PVS-Studio warning: V751 Parameter 'b' is not used inside function body. dh_ameth.c 312

V751: The analyzer detected a suspicious function where one of the parameters is never used while another parameter is used several times. It may indicate an error in the code.

Correct:

```
return ossl_ffc_params_cmp(&a->pkey.dh->params, &b->pkey.dh->params, .....);
```

P.S. It's a [classic](https://pvs-studio.com/en/blog/posts/cpp/0509/) bug. :)

**Comment by t8m** [comment]
Fortunately this affects only legacy DH  PKEYs (which is probably the reason it was not found earlier).

**Comment by t8m** [comment]
Fix in #16568

**Comment by paulidale** [comment]
For reference: [the blog post](https://karpov2007.medium.com/everybody-makes-mistakes-when-writing-comparison-functions-9cd49ea35088).
